bundler/lib/bundler/definition.rb

```diff
@@ -738,7 +738,7 @@ def converge_sources
         # has to be done separately, because we want to keep the locked checksum
         # store for a source, even when doing a full update
         if @locked_gems && locked_source = @locked_gems.sources.find {|s| s == source }
-          source.checksum_store&.use(locked_source.checksum_store)
+          source.checksum_store&.merge(locked_source.checksum_store)
```

Comment:
This should be merge!, but also this seems weird because if source.checksum_store is nil then we should make it not nil by setting the locked_source checksums into it.

```suggestion
          source.checksum_store&.merge!(locked_source.checksum_store)
```